presto-main/src/main/java/com/facebook/presto/operator/FilterAndProjectOperator.java

```diff
@@ -36,15 +36,20 @@
     private final LocalMemoryContext outputMemoryContext;
 
     private final PageProcessor processor;
-    private PageProcessorOutput currentOutput = EMPTY_PAGE_PROCESSOR_OUTPUT;
+    private MergingPageOutput currentOutput;
```

Comment:
`currentOutput` doesn't change through lifetime of the operator (so it's always current and the name is meaningless). Consider renaming it to `mergingOutput`.